### PR TITLE
integrate `wasmcloud-control-interface` into the repository

### PR DESCRIPTION
**Pull request by rvolosatovs**
## Feature or Problem

Integrate https://github.com/wasmCloud/control-interface-client into the repo. While working on that, I have removed a bunch of unused dependencies and even unused code.

Funny enough, the `ring` dependency in this crate was blocking us from implementing #724, but it was never actually used by the crate

### CODE REVIEW

File: .github/workflows/wasmcloud.yml
Comment (brooksmtownsend) [inline .github/workflows/wasmcloud.yml:501]: nit: Now that we have the dry run, we could restrict the publish to only happen on the tag push right?

Does this action not being gated by tags mean we're just going to release new versions of the crate on merge? I don't think we've designed our workflow for that

File: .github/CODEOWNERS
Comment (brooksmtownsend) [inline .github/CODEOWNERS:31]:
```suggestion
crates/control-interface @wasmCloud/control-interface-maintainers
```

File: Cargo.toml
Comment (brooksmtownsend) [inline Cargo.toml:171]: The reason why we explicitly versioned these from `*` was so that crates.io would publish dependents and know what version to use, does using `version = "0"` let this happen? If I remember correctly the dry run didn't catch it before

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
Requests for process reasons but the crate move looks good

File: .github/workflows/wasmcloud.yml
Comment (rvolosatovs) [reply .github/workflows/wasmcloud.yml:501]: Crates are only published when the matching tag is pushed

File: Cargo.toml
Comment (rvolosatovs) [reply Cargo.toml:171]: It should, but there's really only one way to find out if dry run does not catch it

**Review by brooksmtownsend — APPROVED** [review]
Once we merge and attempt to cut a tag, let's see if specifying 0 works for us. If it's the semver 0 major version then I would assume that works

**Pull request merged**
